client/image_prune.go

```diff
@@ -22,8 +23,13 @@ func (cli *Client) ImagesPrune(ctx context.Context, pruneFilters filters.Args) (
 		return report, err
 	}
 
-	serverResp, err := cli.post(ctx, "/images/prune", query, nil, nil)
-	if err != nil {
+	cCtx, cancel := context.WithCancel(ctx)
+	RegisterSignalHandler(func() {
+		cancel()
+	})
```

Comment:
Just realized I can't actually get an http answer if I cancel the context, so I'll revert this part of the changes and just print the changes into the daemon log.